index.js

```diff
@@ -20,19 +23,22 @@ function File(file) {
     file = {};
   }
 
-  // Record path change
-  var history = file.path ? [file.path] : file.history;
-  this.history = history || [];
-
-  this.cwd = file.cwd || process.cwd();
-  this.base = file.base || this.cwd;
-
   // Stat = files stats object
   this.stat = file.stat || null;
 
   // Contents = stream, buffer, or null if not read
   this.contents = file.contents || null;
 
+  // Replay path history to ensure proper normalization and trailing sep
+  var history = file.path ? [file.path] : file.history || [];
+  this.history = [];
+  history.forEach(function(path) {
+    self.path = path;
```

Comment:
If stats aren't passed into the constructor, this normalization might break things.  How do you feel about only doing this if we have `this.stat`?